signac/core/collection_api.py

```diff
@@ -18,7 +22,7 @@
 try:
     from collections.abc import Collection
 except ImportError:
-    # create Collection
+    # Collection does not exist in Python 3.5, only Python 3.6 or newer.
```

Comment:
I think it's ok to make the decision to develop this feature for Python 3.6+. Python 3.5 is almost EOL and major libraries like numpy have already dropped support in their latest releases. @glotzerlab/signac-committers

Reply:
@csadorf I support that, and as you mentioned, it is in line with the drop schedule here: https://numpy.org/neps/nep-0029-deprecation_policy.html.

Reply:
@vishav1771 To drop Python 3.5, we'll have to do a few things (drop CI tests, update documentation, etc.). It's fine to leave this shim in here for Python 3.5 for the moment, it won't hurt anything. We can drop support on `master` and merge it into this branch before finalizing your PR.